src/components/People.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { Link, useLocation, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+
+import { usePeopleState } from '../store/PeopleContext';
+
+import { Person } from '../types';
+
+export const People: React.FC = () => {
+  const { people } = usePeopleState();
+
+  const { pathname } = useLocation();
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || "";
+  const q = searchParams.get('q') || "";
+  const centuries = Number(searchParams.get('centuries') || 0);
+
+  const getSlugByName = (name: string) => {
+    const human = people.find((person: Person) => person.name === name);
+
+    return human?.slug || null;
+  };
+
+  const setNameColor = (sex: string) => {
+    if (null) {
+      return;
+    }
+
+    return cn({
+      'has-text-danger': sex === 'f',
+      'has-text-success': sex === 'm',
+    });
+  };
+
+  const getFilteredPeople = (people: Person[]) => {
+    if(sex === 'm') {
+      return people
+        .filter((person: Person) => person.sex === 'm')
+    } else if(sex === 'f') {
+      return people
+        .filter((person: Person) => person.sex === 'f')
+    } else if (q) {
+      return people
```

Comment:
```suggestion
    }
    
    if (q) {
      filteredPeople = people
```
it looks redundant to combine all these parts with else block, and return in each part

Reply:
yep, bro, there is nothing to check here, that is very crude version